common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationFactory.java

```diff
@@ -1130,24 +1146,30 @@ public static String[] getICULocaleNames() {
 
   public static UTF8String getCollationKey(UTF8String input, int collationId) {
```

Comment:
let's avoid making unnecessary `applyTrimmingPolicy` calls here - even though it's functionally a noop, we should short-circuit if possible

Reply:
`if (...) input = applyTrimmingPolicy(input, ...);`